packages/container/tests/container_test.js

```diff
@@ -615,3 +616,44 @@ QUnit.skip('#factoryFor does not add properties to the object being instantiated
   // not via registry/container shenanigans
   assert.deepEqual(Object.keys(instance), []);
 });
+
+if (isFeatureEnabled('ember-module-unification')) {
+  QUnit.module('Container module unification');
+
+  QUnit.test('The container can pass a source to factoryFor', function() {
+    let PrivateComponent = factory();
+    let src = 'template:routes/application';
+    let registry = new Registry();
+    registry.resolve = function(fullName, { source }) {
+      if (fullName === 'component:my-input' && src === source) {
+        return PrivateComponent;
+      }
+    };
+
+    let container = registry.container();
+
+    strictEqual(container.factoryFor('component:my-input', { source: src }).class, PrivateComponent, 'The correct factory was provided');
+    strictEqual(container.factoryManagerCache[`template:routes/application:component:my-input`].class,
```

Comment:
Instead of asserting the contents of the private cache, can the cache be make `_` private and here you can assert `resolve` or the method on the resolver isn't called a second time? or that it *is* called with a different `source`? Test the behavior, not the state of the cache?